### PR TITLE
k8s: fix GetSecret for non-existent secret, ErrInvalidSecretId

### PR DESCRIPTION
**Pull request by baum**
**What this PR does / why we need it**:

    k8s: fix GetSecret for non-existent secret, ErrInvalidSecretId

    - GetSecret returns ErrInvalidSecretId, when secret not found.
    - Fix k8s handling of k8s secret "not found" path
  

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: k8s/k8s.go [hunk 1]
```diff
@@ -35,12 +37,12 @@ func (s *k8sSecrets) GetSecret(
 
 	secret, err := core.Instance().GetSecret(secretName, namespace)
 	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
```
Comment (leseb) [inline k8s/k8s.go:40]: How about using `kerrors.IsNotFound(err)` Where `kerrors` comes from `kerrors "k8s.io/apimachinery/pkg/api/errors"`. Perhaps `core` has a similar function?

**Review by leseb — CHANGES_REQUESTED** [review]

File: k8s/k8s.go (hunk 1 above)
Comment (baum) [reply k8s/k8s.go]: Even better 😊, pushed `kerrors.IsNotFound(err)` and unit test.

Comment (baum) [reply k8s/k8s.go]: Are github actions disabled on this repository?

File: .github/workflows/k8s-test.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  run-k8s-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: deploy kind cluster
+        run: |
+          tests/scripts/install-5nodes-kind-cluster.sh
```
Comment (leseb) [inline .github/workflows/k8s-test.yml:15]: why do we need a cluster to run the unit test? Seems overkill. Also unit tests run already with the TravisCI job so I'm not if this code addition is needed.

File: k8s/k8s.go [hunk 3]
```diff
@@ -35,12 +36,12 @@ func (s *k8sSecrets) GetSecret(
 
 	secret, err := core.Instance().GetSecret(secretName, namespace)
 	if err != nil {
+		if kerrors.IsNotFound(err) {
```
Comment (leseb) [inline k8s/k8s.go:39]: Doesn't `"github.com/portworx/sched-ops/k8s/core"` provide the `IsNotFound()` function? It will be good to avoid a new dependency.

**Review by leseb — CHANGES_REQUESTED** [review]

File: .github/workflows/k8s-test.yml (hunk 2 above)
Comment (baum) [reply .github/workflows/k8s-test.yml:15]: This is basically an integration test, I don't mind axing it.  What is the alternative to running a k8s cluster? How to test a way in which cluster reacts to accessing a secret that does not exist?   Could this functionality be tested with TravisCI? Please share your thoughts.

Comment (leseb) [reply .github/workflows/k8s-test.yml:15]: We might be able to just use a unit test for this. I'm not sure how they are articulated but if we can mock a server response then we can test that a secret does not exist. Also, it looks like GitHub action needs to be approved by a maintainer and this has never been the case so far....

File: k8s/k8s.go (hunk 3 above)
Comment (baum) [reply k8s/k8s.go:39]: Where?

The secrets update logic uses `strings.Contains(err.Error(), "not found"` to detect this condition, see [here](https://github.com/portworx/sched-ops/blob/1f213fafc48e8649820ebf683c4d23ce0fc28d99/k8s/core/secrets.go#L75)

File: k8s/k8s.go (hunk 1 above)
Comment (leseb) [reply k8s/k8s.go]: Not really but a maintainer must run the job manually first since you are a first-time contributor

File: k8s/k8s.go (hunk 3 above)
Comment (leseb) [reply k8s/k8s.go:39]: Ok so they don't have it. I'm sure how the maintainers feel about adding another dependency. At the same time the initial version with `strings.Contains(err.Error(), "not found"` is probably ok and not worth the additional dep.

Comment (baum) [reply k8s/k8s.go]: Sounds good, pushing the string detection version. This is the minimal change to fix the logic.

**Review by leseb — APPROVED** [review]
LGTM!

File: k8s/k8s.go (hunk 1 above)
Comment (adityadani) [reply k8s/k8s.go:40]: I don't see this `kerrors.IsNotFound(err)` change in the latest version of PR. Btw I have triggered the github actions on this PR.

File: .github/workflows/k8s-test.yml (hunk 2 above)
Comment (adityadani) [reply .github/workflows/k8s-test.yml]: Did we remove this code out in the recent version of this PR?

File: k8s/k8s.go (hunk 3 above)
Comment (adityadani) [reply k8s/k8s.go]: It should be ok to add a new dependency. Better to have a strict error check instead of string comparison.

Comment (leseb) [reply k8s/k8s.go]: Not sure if the new dep is worth it for such a small check though...

**Review by adityadani — APPROVED** [review]

**Pull request closed without merging**